parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -95,14 +97,32 @@ installSGXDrivers() {
 }
 
 installContainerRuntime() {
-    if [[ "$CONTAINER_RUNTIME" == "docker" ]]; then
-        installMoby
-    fi
     {{if NeedsContainerd}}
-        installContainerd
+        installStandaloneContainerd
+    {{else}}
+        installMoby
     {{end}}
 }
 
+{{if NeedsContainerd}}
+# CSE+VHD can dicate the containerd version, users don't care as long as it works
+installStandaloneContainerd() {
+    # azure-built runtimes have a "+azure" suffix in their version strings (i.e 1.4.1+azure). remove that here.
+    CURRENT_VERSION=$(containerd -version | cut -d " " -f 3 | sed 's|v||' | cut -d "+" -f 1)
+    # v1.4.1 is our lowest supported version of containerd
+    local CONTAINERD_VERSION="1.4.1"
+    if semverCompare ${CURRENT_VERSION:-"0.0.0"} ${CONTAINERD_VERSION}; then
+        echo "currently installed containerd version ${CURRENT_VERSION} is greater than (or equal to) target base version ${CONTAINERD_VERSION}. skipping installStandaloneContainerd."
+    else
```

Comment:
Do we need to removeMoby in this case as well?

Reply:
We want the cse to be inter-operable for all AKS VHD SKUs so we should have a removeMoby call in the case we run this on a Docker VHD instead of a containerd VHD. In theory rp/agentbaker should have logic that would block that case but it's better that it works in that case anyways. 
I did some manual testing of running CSE scripts templated with containerd runtime on a vanilla docker VHD and it works.